src/include/sof/audio/component_ext.h

```diff
@@ -147,13 +148,46 @@ static inline int comp_trigger_remote(struct comp_dev *dev, int cmd)
 	return idc_send_msg(&msg, IDC_BLOCKING);
 }
 
+static inline int comp_trigger_local(struct comp_dev *dev, int cmd)
+{
+	int ret;
+
+	ret = dev->drv->ops.trigger(dev, cmd);
+
+	/* start a thread in case of shared component or DP scheduling */
+	if (dev->task) {
+		/* schedule or cancel task, component must be shared or in DP domain */
+		assert(dev->is_shared ||
+		       dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP);
```

Comment:
sorry, don't understand: why does it have to be shared?

Reply:
its folllwing (in fact verifying checking) the logic that was here before
Note that IDC on second core will finally call comp_trigger (and comp_trigger_local on second core)

amd the condition was:
return (**dev->is_shared && !cpu_is_me(dev->ipc_config.core))** ? comp_trigger_remote(dev, cmd) : dev->drv->ops.trigger(dev, cmd);

And according to description (before/after)
```
        struct task *task; /**< component's processing task used only
                                   * for components running on different core
                                 * than the rest of the pipeline

```
```
        struct task *task;	/**< component's processing task used
			  *  1) for components running on different core
			  *    than the rest of the pipeline
			  *  2) for all DP tasks
			  *
```
may be removed, it is just double check

Reply:
@marcinszkudlinski correct me if I'm wrong: this function is entered from `comp_trigger()` under the condition `if (!dev->is_shared || cpu_is_me(dev->ipc_config.core))` i.e. one of the conditions when this function is entered is a normal local component trigger with the LL scheduler, which now will trigger this your assert. It really looks both unneeded and wrong to me.

Reply:
the assert won't be met **only** in case there's **a task** assigned to a component that is
 - not DP **AND** is running on the same core that the rest of pipeline

it won't be hit for any component that have no task assigned ( task == NULL) (line 158 above)

It is following the description of *task field I stated above, is the description incorrect?
Agreed it is jus ta double check, I can remove it, no problem

Reply:
ah, now I understand, I completely missed the point with `dev->task != NULL`. So in fast - yes, the indication, that we can only land here either with a shared component or with the DP scheduler would be useful here - either as a comment, or an `if` with a `comp_err(); return;` or as an assertion, but not a blocker of course.